python/ray/experimental/channel/common.py

```diff
@@ -239,11 +250,31 @@ def get_num_reads(self) -> int:
     def start(self):
         raise NotImplementedError
 
-    def _read_list(self) -> Any:
+    def _read_list(self, timeout: Optional[float] = None) -> List[Any]:
+        """
+        Read a list of values from this reader.
+
+        Args:
+            timeout: The maximum time in seconds to wait for reading.
+                If this is 0, either return the result immediately
+                (if already available) or raise an exception immediately.
+                Must be non-negative.
+
+        """
         raise NotImplementedError
 
-    def read(self) -> List[Any]:
-        outputs = self._read_list()
+    def read(self, timeout: Optional[float] = None) -> List[Any]:
+        """
+        Read from this reader.
+
+        Args:
+            timeout: The maximum time in seconds to wait for reading.
```

Comment:
Why don't we support -1 for infinity? I think we need this case for the intermediate actors, which should never time out while reading their inputs.

Edit: Oh is the convention `None` for infinity? If so, that sounds good, but please document it.

Reply:
Yeah I used `None` for infinity. It follows the same convention as `ray.get()` which there are also shared code paths with us. Also followed the same convention to convert `None` to -1 to be consistent with Ray:
https://github.com/ray-project/ray/blob/755a49bbc2345650ff90d4706d9c69f87b8a77bd/python/ray/_private/worker.py#L847

I think `None` also makes sense to indicate no timeout == infinite timeout.